lib/webpack/rails/helper.rb

```diff
@@ -18,9 +18,11 @@ def webpack_asset_paths(source, extension: nil)
         paths = Webpack::Rails::Manifest.asset_paths(source)
         paths = paths.select {|p| p.ends_with? ".#{extension}" } if extension
 
-        host = ::Rails.configuration.webpack.dev_server.host
         port = ::Rails.configuration.webpack.dev_server.port
 
+        host = ::Rails.configuration.webpack.dev_server.host
+        host = instance_eval(&host) if host.respond_to?(:call)
```

Comment:
The idea is that the proc is evaluated in the context of the view – so you can use the `request` object and any view helpers you wan.